breathecode/certificate/tests/urls/tests_root.py

```diff
@@ -0,0 +1,235 @@
+        }]
+        response = self.client.post(url, data, format='json')
+        json = response.json()
+        expected = {
+            'detail': "You (user: 1) don't have this capability: crud_certificate for academy 1",
+            'status_code': 403
+        }
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+    
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_certificate_re_attemps_without_cohort_user(self):
+        """Test /root with auth"""
+        """ No cohort_user for the request"""
+        self.headers(academy=1)
+        model = self.generate_models(authenticate=True, cohort=True, user=True,
+            profile_academy=True, role="STUDENT", capability='crud_certificate')
+
+        url = reverse_lazy('certificate:root')
+        data = [{
+            'cohort_slug':  model['cohort'].slug,
+            'user_id':  model['user'].id,
+        }]
+        response = self.client.post(url, data, format='json')
+        json = response.json()
+        expected = {
+            'detail': 'student-not-found-in-cohort',
+            'status_code': 404
+        }
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+
+
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_certificate_re_attemps_without_user_specialty(self):
+        """Test /root with auth"""
+        """ No user_specialty for the request"""
+        self.headers(academy=1)
+        model = self.generate_models(authenticate=True, cohort=True, user=True,
+            profile_academy=True, role="STUDENT", capability='crud_certificate', 
+            cohort_user=True)
+
+        url = reverse_lazy('certificate:root')
+        data = [{
+            'cohort_slug':  model['cohort'].slug,
+            'user_id':  model['user'].id,
+        }]
+        response = self.client.post(url, data, format='json')
+        json = response.json()
+        expected = {
+            'detail': 'no-user-specialty',
+            'status_code': 404
+        }
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(self.all_user_specialty_dict(), [])
+
+    
+    @patch(GOOGLE_CLOUD_PATH['client'], apply_google_cloud_client_mock())
+    @patch(GOOGLE_CLOUD_PATH['bucket'], apply_google_cloud_bucket_mock())
+    @patch(GOOGLE_CLOUD_PATH['blob'], apply_google_cloud_blob_mock())
+    def test_certificate_re_attemps(self):
```

Comment:
We need a new test like it that generate two UserSpecialty, to certificate the view accept bulk mode